### PR TITLE
L95: Python reflection client

### PR DESCRIPTION
**Pull request by tomerv**

### CODE REVIEW

File: L95-python-reflection-client.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+Python Reflection Client
+----
+* Author: Tomer Vromen
+* Approver: a11r, lidizheng (proposed)
+* Status: Draft
+* Implemented in: Python
+* Last updated: 2022-01-20
+* Discussion at: (TBD, filled after thread exists)
```
Comment (lidizheng) [inline L95-python-reflection-client.md:8]: Can you post a thread in grpc-io group?

File: L95-python-reflection-client.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+Python Reflection Client
+----
+* Author: Tomer Vromen
+* Approver: a11r, lidizheng (proposed)
```
Comment (lidizheng) [inline L95-python-reflection-client.md:4]: Let's add @gnossen, remove a11r

File: L95-python-reflection-client.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+## Proposal
+
+Provide a Python implementation for client-side reflection, modeled after the existing C++ implementation.
+
+* Implement `ProtoReflectionDescriptorDatabase` which implements the
+[`DescriptorDatabase`](https://googleapis.dev/python/protobuf/latest/google/protobuf/descriptor_database.html#google.protobuf.descriptor_database.DescriptorDatabase)
+interface.
+* Write tests.
+* Write documentation.
```
Comment (lidizheng) [inline L95-python-reflection-client.md:39]: You may want to update https://github.com/grpc/grpc/blob/master/doc/python/sphinx/grpc_reflection.rst to add the new APIs.

**Review by lidizheng — APPROVED** [review]
This is a really nice tool to have for gRPC clients. LGTM.

But before merging the proposal, we might want to create the discussion thread, and wait 10 days (comment period).

File: L95-python-reflection-client.md (hunk 1 above)
Comment (tomerv) [reply L95-python-reflection-client.md]: done

File: L95-python-reflection-client.md (hunk 2 above)
Comment (tomerv) [reply L95-python-reflection-client.md]: done.

File: L95-python-reflection-client.md (hunk 3 above)
Comment (tomerv) [reply L95-python-reflection-client.md:39]: good idea, will do that

**Comment by tomerv** [comment]
Not sure what to do with the CLA, seems like it's stuck. Should I start a new pull request?

**Comment by lidizheng** [comment]
I think the check has passed. See the lower section of the GitHub UI. That comment just... stale.

File: L95-python-reflection-client.md (hunk 3 above)
Comment (tomerv) [reply L95-python-reflection-client.md:39]: Done.

**Pull request merged**